common/workunit/workunit.ipp

```diff
@@ -734,6 +736,45 @@ protected:
     unsigned id;
 };
 
+class WorkUnitWaiter : public CInterface, implements ISDSSubscription, implements IAbortHandler
+{
+    Semaphore changed;
+    SubscriptionId change;
+public:
+    IMPLEMENT_IINTERFACE;
+
+    WorkUnitWaiter(const char *xpath)
+    {
+        change = querySDS().subscribe(xpath, *this, false);
+        aborted = false;
+    }
+    ~WorkUnitWaiter()
+    {
+        assertex(change==0);
```

Comment:
I know not new, but I suspect more likely to cause problems than help, e.g. if process is winding up after exception and happens not to have unsubscribed, or unsubscribe causes exeception than WorkUnitWaiter is destroyed, it will cause this assert to be hit and likely cause core.